packages/docid/src/__tests__/doc-id.test.ts

```diff
@@ -0,0 +1,210 @@
+const DOC_ID_URL = 'ceramic://kjzl6cwe1jw147dvq16zluojmraqvwdmbh61dx9e0c59i344lcrsgqfohexp60s';
+const DOC_ID_BYTES = multibase.decode(DOC_ID_STRING);
+const DOC_ID_LEGACY = '/ceramic/kjzl6cwe1jw147dvq16zluojmraqvwdmbh61dx9e0c59i344lcrsgqfohexp60s';
+
+const DOC_ID_WITH_COMMIT =
+  'k1dpgaqe3i64kjqcp801r3sn7ysi5i0k7nxvs7j351s7kewfzr3l7mdxnj7szwo4kr9mn2qki5nnj0cv836ythy1t1gya9s25cn1nexst3jxi5o3h6qprfyju';
+const DOC_ID_WITH_COMMIT_BYTES = multibase.decode(DOC_ID_WITH_COMMIT);
+const DOC_ID_WITH_0_COMMIT = 'k3y52l7qbv1frxwipl4hp7e6jlu4f6u8upm2xv0irmedfkm5cnutmezzi3u7mytj4';
+const DOC_ID_WITH_0_COMMIT_BYTES = multibase.decode(DOC_ID_WITH_0_COMMIT);
+const DOC_ID_WITH_COMMIT_LEGACY =
+  '/ceramic/kjzl6cwe1jw147dvq16zluojmraqvwdmbh61dx9e0c59i344lcrsgqfohexp60s?commit=bagjqcgzaday6dzalvmy5ady2m5a5legq5zrbsnlxfc2bfxej532ds7htpova';
+const DOC_ID_WITH_0_COMMIT_LEGACY = '/ceramic/kjzl6cwe1jw147dvq16zluojmraqvwdmbh61dx9e0c59i344lcrsgqfohexp60s?commit=0';
+
+describe('constructor', () => {
+  test('create by parts (type:int, cid:cid)', () => {
+    const type = 0;
+    const cid = new CID(CID_STRING);
+    const docid = new DocID(type, cid);
+
+    expect(docid.type).toEqual(type);
+    expect(docid.cid.toString()).toEqual(CID_STRING);
+    expect(docid.toString()).toMatchSnapshot();
+  });
+});
+
+describe('.fromBytes', () => {
+  test('create from bytes ', () => {
+    const docid = DocID.fromBytes(DOC_ID_BYTES);
+
+    expect(docid.type).toEqual(0);
+    expect(docid.cid.toString()).toEqual(CID_STRING);
+    expect(docid.toString()).toMatchSnapshot();
+  });
+
+  test('fail to create from garbage bytes', () => {
+    expect(() =>
+      DocID.fromBytes(new CID('bagcqcerakszw2vsovxznyp5gfnpdj4cqm2xiv76yd24wkjewhhykovorwo6a').bytes),
+    ).toThrow();
+  });
+
+  test('create from bytes including commit', () => {
+    const docid = DocID.fromBytes(DOC_ID_WITH_COMMIT_BYTES);
+
+    expect(docid.type).toEqual(0);
+    expect(docid.cid.toString()).toEqual(CID_STRING);
+    expect(docid.toString()).toMatchSnapshot();
+  });
+
+  test('create from bytes including commit 0', () => {
+    const docid = DocID.fromBytes(DOC_ID_WITH_0_COMMIT_BYTES);
+
+    expect(docid.type).toEqual(0);
+    expect(docid.cid.toString()).toEqual(CID_STRING);
+    expect(docid.toString()).toMatchSnapshot();
+  });
+
+  test('roundtrip docID bytes', () => {
+    const docid = new DocID('tile', CID_STRING);
+    const docid2 = DocID.fromBytes(docid.bytes);
+    expect(docid.toString()).toEqual(docid2.toString());
+  });
+});
+
+describe('.fromString', () => {
+  test('create from string', () => {
+    const docid = DocID.fromString(DOC_ID_STRING);
+
+    expect(docid.type).toEqual(0);
+    expect(docid.cid.toString()).toEqual(CID_STRING);
+    expect(docid.toString()).toMatchSnapshot();
+  });
+
+  test('create from string including commit', () => {
+    const docid = DocID.fromString(DOC_ID_WITH_COMMIT);
```

Comment:
It feels like maybe we should throw in this case?

Reply:
should.